src/lib/db/access-store.ts

```diff
@@ -46,13 +50,79 @@ interface IPermissionRow {
     role_id: number;
 }
 
+type ResolvedPermission = {
+    id: number;
+    name?: string;
+    environment?: string;
+};
+
 export class AccessStore implements IAccessStore {
     private logger: Logger;
 
     private timer: Function;
 
     private db: Db;
 
+    // for internal usage only
+
+    private isIdPermissionRef = (permission: PermissionRef): boolean => {
+        return (permission as IdPermissionRef).id !== undefined;
+    };
+
+    private permissionNamesToIds = async (
+        permissions: NamePermissionRef[],
+    ): Promise<ResolvedPermission[]> => {
+        if (permissions === undefined || permissions.length === 0) {
+            return [];
+        }
+        const permissionNames = permissions
+            .filter((p) => p.name !== undefined)
+            .map((p) => p.name);
+
+        const rows = await this.db
+            .select('id as permissionId', 'permission')
+            .from(T.PERMISSIONS)
+            .whereIn('permission', permissionNames);
+
+        return rows.map((row) => ({
+            id: row.permissionId as number,
+            name: row.permission as string,
+            environment: permissions.find(
+                (p) => p.name === (row.permission as string),
+            )?.environment,
```

Comment:
I don't like the performance of this... a find inside a map is not ideal

Reply:
Is it a problem? O(n*m) isn't an issue when n and m are small. Should be easy enough to build a Map earlier in the function and get O(1) lookups

Reply:
Exactly what I had in mind

Reply:
Assume positive intentions here, I'm struggling with formulating the question. 
Why are we fetching ids again? Wasn't the goal to change to names as the primary key?

Reply:
> Wasn't the goal to change to names as the primary key?

Yes, but our OpenAPI spec states that ids are required while names are optional. Adding the name as required is a breaking change having to wait until the next major release. **I think** because of this reason we can't drop support for id's (customers might be using them already).

Reply:
I think this is correct, we can trade off some performance for correctness for a while. When we make a breaking change we can clean this up